analysers/trials/throughput.py

```diff
@@ -39,25 +39,30 @@ def computeThroughput(dataRDD):
                 largest = t
     
     if largest is None or smallest is None:
-        return (None, None)
+        return None
     
     delta = largest - smallest
     delta = delta.total_seconds()
     
     tp = len(data)/float(delta)
     
-    return (tp, delta)
+    return tp
 
-def createQuery(sc, dataRDD, experimentID, trialID, nToIgnore):
-    from commons import cutNInitialProcesses
-    
-    data = cutNInitialProcesses(dataRDD, nToIgnore)
-        
-    dataRDD = sc.parallelize(data)
+def createQuery(sc, dataRDD, experimentID, trialID):
+    queries = []
     
     tp = computeThroughput(dataRDD)
     
-    return [{"experiment_id":experimentID, "trial_id":trialID, "throughput":tp[0], "execution_time":tp[1]}]
+    queries.append({"experiment_id":experimentID, "trial_id":trialID, "process_definition_id":"all", "throughput":tp})
+    
+    processes = dataRDD.map(lambda a: a["process_definition_id"]).distinct().collect()
+    
+    for process in processes:
+        tp = computeThroughput(dataRDD.filter(lambda r: r['process_definition_id'] == process))
```

Comment:
You should just use the `execution time` and the `number of process instance` metrics, not compute then again here